File: ruby/IITK_Complexity_3.rb
def power(n)
  if n == 0
    return 1
  end

  res = power(n/2)

  if n%2 == 0
    return res * res
  else
    return res * res * 3
  end
end


if __FILE__ == $0
  n = gets().chomp().to_i()

  puts(power(n))
end

File: java/IITK_Complexity_3.java
import java.util.Scanner;


class Solution {
    public static int power(int n) {
        if (n == 0)
            return 1;

        int res = power(n/2);

        if (n%2 == 0)
            return res * res;
        else
            return res * res * 3;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        System.out.println(power(n));
    }
}
